tests/operations/location-operations.test.tsx

```diff
@@ -0,0 +1,29 @@
+import { SearchIntent, useSearchActions } from '@yext/search-headless-react';
+import { getUserLocation, updateLocationIfNeeded } from '../../src/utils/location-operations';
+
+describe('Location Operations updateLocationIfNeeded', () => {
+  it('fires getUserLocation with NearMe intent', async () => {
+    const searchActions = useSearchActions();
```

Comment:
how is useSearchActions working outside of a react component here?

Reply:
uHHh honestly I'm not sure why it worked lmao

how would you suggest approaching this instead?

Reply:
Would you mind digging a little into what's happening right now? If it works as is then that's good